lib/ts/recipe/thirdpartypasswordless/types.ts

```diff
@@ -0,0 +1,295 @@
+    };
+
+// TODO: move the definition here if necessary
+export type SignInUpFeatureConfigInput = WithRenamedProp<
+    PWLessSignInUpFeatureConfigInput,
+    "emailOrPhoneFormStyle",
+    "providerAndEmailOrPhoneFormStyle"
+> & {
+    providers?: (Provider | CustomProviderConfig)[];
+};
+
+export type UserInput = (
+    | {
+          contactMethod: "EMAIL";
+
+          validateEmailAddress?: (email: string) => Promise<string | undefined> | string | undefined;
+
+          signInUpFeature?: SignInUpFeatureConfigInput;
+      }
+    | {
+          contactMethod: "PHONE";
+
+          validatePhoneNumber?: (phoneNumber: string) => Promise<string | undefined> | string | undefined;
+
+          signInUpFeature?: SignInUpFeatureConfigInput & {
+              /*
+               * Must be a two-letter ISO country code (e.g.: "US")
+               */
+              defaultCountry?: CountryCode;
+          };
+      }
+    | {
+          contactMethod: "EMAIL_OR_PHONE";
+
+          validateEmailAddress?: (email: string) => Promise<string | undefined> | string | undefined;
+          validatePhoneNumber?: (phoneNumber: string) => Promise<string | undefined> | string | undefined;
+
+          signInUpFeature?: SignInUpFeatureConfigInput & {
+              defaultCountry?: CountryCode;
+              guessInternationPhoneNumberFromInputPhoneNumber?: (
+                  inputPhoneNumber: string,
+                  defaultCountryFromConfig?: CountryCode
+              ) => Promise<string | undefined> | string | undefined;
+          };
+      }
+) & {
+    override?: {
+        functions?: (
+            originalImplementation: TPPWlessRecipeInterface,
+            builder?: OverrideableBuilder<TPPWlessRecipeInterface>
+        ) => TPPWlessRecipeInterface;
+        components?: ComponentOverrideMap;
+    } & AuthRecipeUserInputOverride;
+    linkClickedScreenFeature?: PasswordlessFeatureBaseConfig;
+    oAuthCallbackScreen?: FeatureBaseConfig;
+    disablePasswordless?: boolean;
+} & AuthRecipeModuleUserInput<GetRedirectionURLContext, PreAPIHookContext, OnHandleEventContext>;
+
+export type Config = UserInput &
+    AuthRecipeModuleConfig<GetRedirectionURLContext, PreAPIHookContext, OnHandleEventContext>;
+
+export type NormalizedSignInUpFeatureConfig = {
+    resendEmailOrSMSGapInSeconds: number;
+    providers?: (Provider | CustomProviderConfig)[];
+    defaultCountry?: CountryCode;
+    guessInternationPhoneNumberFromInputPhoneNumber: (
+        inputPhoneNumber: string,
+        defaultCountryFromConfig?: CountryCode
+    ) => Promise<string | undefined> | string | undefined;
+    privacyPolicyLink?: string;
+    termsOfServiceLink?: string;
+    // TODO: better name
+    providerAndEmailOrPhoneFormStyle: Styles;
+    userInputCodeFormStyle: Styles;
+    linkSentScreenStyle: Styles;
+    closeTabScreenStyle: Styles;
+    disableDefaultImplementation?: boolean;
+};
+
+export type NormalisedConfig = {
```

Comment:
The NormalisedConfig should actually be the input type for passwordless and third-party - cause those will normalise the input in their recipe anyway.

Reply:
@porcellus did you miss this?